code-challenges.test.js

```diff
@@ -14,38 +14,143 @@
-  { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
-  { name: "arthur dent", occupation: "a radio employee" }
-]
-// Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
+describe("betterThanGrammarly", () => {
+  it("takes in an array of objects and returns an array with a sentence about each person with their name capitalized", () => {
+      const people = [
+          { name: "ford prefect", occupation: "a hitchhiker" },
+          { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
+          { name: "arthur dent", occupation: "a radio employee" }
+        ]
+    expect(betterThanGrammarly(people)).toEqual(["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."])
+  })
+})
+
+//  ReferenceError: betterThanGrammarly is not defined
 
 
 // b) Create the function that makes the test pass.
 
+// const betterThanGrammarly = (array) => 
+//   return array.map =(({key1,key2}) => {
+//     return key1.charAt(0).toUpperCase() +
+//   })
+// }
+
+
+//Yall really know how to pick some hard prompts, cause boy..... this is a doozy. I've eaten three plates of cheez-its trying to figure this out. I'm stressed and eating my feelings. 
+
+
+// const betterThanGrammarly = (array) => {
+//   for (let i=0; i<array.length; i++) {
+//     let value = array[i];
+//     for (let key in value) {
+//       if (value.hasOwnProperty(key)) {
+//           return value[key].charAt(0).toUpperCase() + value[key].substring(1) 
+//       }
+//     } 
+//   }
+// }
+
+//I feel like a somehow have to nest a for in loop inside the .map but I could be delious. 
+
+// const betterThanGrammarly = (people) => {
+//   let rows = people.length;
+//   for(let i=0; i<rows; i++) {
+//     for(let prop in people[i]) {
+//       // console.log(people[i][prop])
+//       return people[i][prop].charAt(0).toUpperCase() + people[i][prop].substring(1)
+//     }
+//   }
+// }
+
+// I was delirious. It didnt work 
+
+//---Puesdocode---//
+  //declare a function called betterThanGrammarly 
+  //set param of array
+  //declare a variable called name 
+    // use .split to convert the name key into an array
+  //declare a variable firstName 
+    // use chartAt and toUppercase to captialize the first letter of the first word in the name key array
+  //declare a variable lastName 
+    // use chartAt and toUppercase to captialize the first letter of the second word in the name key array
+  //use string interpolation to return the sentence.
+
+const betterThanGrammarly = (array) => {
+  return array.map(value => {
+    let name = value.name.split(" ")
+    let firstName = name[0].charAt(0).toUpperCase() + name[0].slice(1)
+    let lastName = name [1].charAt(0).toUpperCase() + name [1]. slice(1)
+    return `${firstName} ${lastName} is ${value.occupation}.`
+  })
+}
+
+// Test Suites: 1 passed, 1 total
```

Comment:
Fantastic job, Will!  You gave that problem your all AND have beautiful code to show for it.  Your pseudocode is great, clean code, and passing test - congrats!